HeterogeneousCore/AlpakaUtilities/interface/alpakastdAlgorithm.h

```diff
@@ -0,0 +1,72 @@
+#ifndef AlpakaCore_alpakastdAlgorithm_h
+#define AlpakaCore_alpakastdAlgorithm_h
+
+#include <algorithm>
+#include <functional>
+#include <utility>
+
+#include <alpaka/alpaka.hpp>
+
+// reimplementation of std algorithms able to compile with Alpaka,
+// mostly by declaring them constexpr
+
+namespace alpaka_std {
```

Comment:
@makortel a technical question: would it work (and be allowed) to declare this as `namespace alpaka::std` ?

If we do that, any code inside `namespace alpaka { ... }` that uses `std::lower_bound` would pick which version ?

Reply:
I'd be wary in using `namespace std` even inside another namespace.

> If we do that, any code inside `namespace alpaka { ... }` that uses `std::lower_bound` would pick which version ?

Based on our early attempts to use `cms::alpaka` namespace, and `alpaka::<X>` types there, I guess `std::lower_bound` inside `namespace alpaka { ... }` would resolve to `alpaka::std::lower_bound`, if the header defining it is `#include`d, and otherwise to `::std::lower_bound`. I also guess `#include`ng a header defining anything in `alpaka::std` namespace would cause code in `alpaka` namespace to need to use `::std::` for all `std::` names that are not found in `alpaka::std`.

But within CMSSW we are not going to add names in `alpaka` namespace, are we?

Reply:
> But within CMSSW we are not going to add names in `alpaka` namespace, are we?

No, but I'm wondering in case we push these upstream into Alpaka.